oauth/client.go

```diff
@@ -0,0 +1,84 @@
+// Package oauth provides the OAuth APIs
+package oauth
+
+import (
+	"fmt"
+	stripe "github.com/stripe/stripe-go"
+	"github.com/stripe/stripe-go/form"
+	"net/http"
+)
+
+// Client is used to invoke /oauth and related APIs.
+type Client struct {
+	B   stripe.Backend
+	Key string
+}
+
+// AuthorizeURL builds the OAuth authorize URL.
+func AuthorizeURL(params *stripe.AuthorizeURLParams) string {
+	return getC().AuthorizeURL(params)
+}
+
+// AuthorizeURL builds the OAuth authorize URL.
+func (c Client) AuthorizeURL(params *stripe.AuthorizeURLParams) string {
+	express := ""
+	if stripe.BoolValue(params.Express) {
+		express = "/express"
+	}
+	if params.ClientID == nil {
+		params.ClientID = stripe.String(stripe.ClientID)
+	}
+	if params.ResponseType == nil {
+		params.ResponseType = stripe.String("code")
+	}
+	qs := &form.Values{}
+	form.AppendTo(qs, params)
+	return fmt.Sprintf(
+		"https://connect.stripe.com%s/oauth/authorize?%s",
+		express,
+		qs.EncodeValues(),
+	)
+}
+
+// New creates the OAuthToken using a code after successful redirection back.
+func New(params *stripe.OAuthTokenParams) (*stripe.OAuthToken, error) {
+	return getC().New(params)
+}
+
+// New creates the OAuthToken using a code after successful redirection back.
+func (c Client) New(params *stripe.OAuthTokenParams) (*stripe.OAuthToken, error) {
+	// client_secret is sent in the post body for this endpoint.
+	if stripe.StringValue(params.ClientSecret) == "" {
+		params.ClientSecret = stripe.String(stripe.Key)
+	}
+	if stripe.StringValue(params.GrantType) == "" {
+		params.GrantType = stripe.String("authorization_code")
+	}
```

Comment:
OOC, is there some particular rationale behind providing a parameter default here?

It may be a little more convenient I suppose, but I think our normal convention is just allowing the API request to fail the first time and signal back that `grant_type=authorization_code` should have been passed, at which point the user just corrects their implementation. It's one extra step, but they probably won't ever make the mistake again.

Reply:
Yeah it was all about convenience. Trying to make it so that users of this can pass some basics and we send sensible defaults.